src/azure-cli/azure/cli/command_modules/role/_params.py

```diff
@@ -120,7 +120,7 @@ def load_arguments(self, _):
         c.argument('identifier_uri', help='graph application identifier, must be in uri format')
         c.argument('spn', help='service principal name')
         c.argument('upn', help='user principal name, e.g. john.doe@contoso.com')
-        c.argument('query_filter', options_list=['--filter'], help='OData filter')
+        c.argument('query_filter', options_list=['--filter'], help='OData filter, e.g. displayname eq \'test\' and servicePrincipalType eq \'Application\'')
```

Comment:
It'll be great to mention that the query needs to be quotes, like `--filter "displayname eq 'test'"`.